python/equistore-operations/equistore/operations/sort.py

```diff
@@ -0,0 +1,183 @@
+    sort_block_dims_names: bool = True, # sort each block's labels' names lexicographical
+    sort_block_dims_values: bool = True, # sort each block's values' according to the labels
+    descending: bool = False,
+    stable: bool = False
+    # numpy
+    # using kind='quicksort' if False and kind='stable' if True
+) -> TensorMap:
+    """
+    >>> import numpy as np
+    >>> import equistore
+    >>> from equistore import TensorBlock, TensorMap, Labels
+    >>> np.random.seed(1)
+    >>> block = TensorBlock(
+    ...     values=np.arange(9).reshape(3,3),
+    ...     samples=Labels(["structures", "centers"], np.array([[0,3], [0,1], [0,2]])),
+    ...     components=[],
+    ...     properties=Labels(["n", "l"], np.array([[2,0], [3,0], [1,0]])),
+    ... )
+    >>> keys = Labels(names=["key"], values=np.array([[0]]))
+    >>> print(block)
+    TensorBlock
+        samples (3): ['structures', 'centers']
+        components (): []
+        properties (3): ['n', 'l']
+        gradients: None
+    >>> block_sorted = equistore.sort_block(block, sort_block_dims_values=False)
+    >>> block_sorted # order within a label entry has been sorted lexicographical
+    TensorBlock
+        samples (3): ['centers', 'structures']
+        components (): []
+        properties (3): ['l', 'n']
+        gradients: None
+    >>> block_sorted.samples
+    Labels([(3, 0), (1, 0), (2, 0)],
+           dtype=[('centers', '<i4'), ('structures', '<i4')])
+    >>> block.values
+    array([[0, 1, 2],
+           [3, 4, 5],
+           [6, 7, 8]])
+    >>> block_sorted.values # but not the values
+    array([[0, 1, 2],
+           [3, 4, 5],
+           [6, 7, 8]])
+    >>> block_sorted_stepwise = equistore.sort_block(block, block_dims=["properties"])
+    >>> block_sorted_stepwise.values # properties are are sorted which are columns in this case
+    array([[2, 0, 1],
+           [5, 3, 4],
+           [8, 6, 7]])
+    >>> block_sorted_stepwise = equistore.sort_block(block_sorted_stepwise , block_dims=["samples"])
+    >>> block_sorted_stepwise.values # samples are are sorted which are rows in this case
+    array([[5, 3, 4],
+           [8, 6, 7],
+           [2, 0, 1]])
+    >>> np.all(equistore.sort_block(block).values == block_sorted_stepwise.values) # doing everything at once and verify correctness
+    True
+    """
+
+    blocks = []
+    for block in tensor.blocks():
+        blocks.append(
+            sort_block(
+                block=block, block_dims=block_dims, sort_metadata=sort_metadata, descending=descending, stable=stable
+            )
+        )
+    return TensorMap(tensor.keys, blocks)
+
+
+def sort_block(
+    block: TensorBlock,
+    block_dims: List[str] = None,
+    sort_block_dims_names: bool = True, # sort each block's labels' names lexicographical
+    sort_block_dims_values: bool = True, # sort each block's values' according to the labels
+    descending: bool = False,
+    stable: bool = False
```

Comment:
We should not need `stable` here since this is always applied on Labels, which do not contain duplicated rows.

Reply:
yes!